### PR TITLE
test: add test for get-unique-words

### PR DESCRIPTION
**Pull request by ChristieRobson**
https://github.com/knaxus/problem-solving-javascript/issues/143

### CODE REVIEW

**Comment by TheSTL** [comment]
Hi @ChristieRobson your commits are not displayed, setup local git properly and push again.

**Comment by ChristieRobson** [comment]
> Hi @ChristieRobson your commits are not displayed, setup local git properly and push again.

@TheSTL similar to the other PR I am unsure what you need me to set up differently. I can see one commit and 1 file change in this PR. I can also see them if viewing this PR without being logged into my account so I don't think there is a permissions issue. Please advise. 

image

**Comment by TheSTL** [comment]
You commits are not linked with your GitHub account .
For more details go to this link https://help.github.com/en/github/committing-changes-to-your-project/why-are-my-commits-linked-to-the-wrong-user

**Comment by ChristieRobson** [comment]
> You commits are not linked with your GitHub account .
> For more details go to this link https://help.github.com/en/github/committing-changes-to-your-project/why-are-my-commits-linked-to-the-wrong-user
> I think it will work https://stackoverflow.com/questions/26004587/git-commits-are-not-getting-linked-with-my-github-account

I see what you mean. I've updated, should be OK now.

File: src/_DataStructures_/Trees/Trie/get-unique-words/get-unique-words.test.js [hunk 1]
```diff
@@ -11,6 +11,52 @@ describe('Data Structure : Trie : Get unique words', () => {
...
+    expect(() => {
+      getUniqueWords([]);
+    }).toThrow('Cannot read property \'0\' of undefined');
+  });
+
+
+  it('passing an array not in a Trie will throw an error ', () => {
+    expect(() => {
+      getUniqueWords(['bed', 'ball', 'apple']);
+    }).toThrow('Cannot read property \'0\' of undefined');
```
Comment (ashokdey) [inline src/_DataStructures_/Trees/Trie/get-unique-words/get-unique-words.test.js]: Thanks for highlighting, this case was something not handled by me in the implementation. Let me add an error message or a better approach is to return an empty array. 

Accordingly, you will have to take a pull of master and run the tests again, these selected tests will fail and you will just have to update the test cases with the result. 

Thank you again.

Comment (ashokdey) [reply src/_DataStructures_/Trees/Trie/get-unique-words/get-unique-words.test.js]: Code will throw **"Invalid argument: Root of Trie is required"**, please make changes accordingly. Do not forget to take the pull of master: **git pull origin master**

Comment (ChristieRobson) [reply src/_DataStructures_/Trees/Trie/get-unique-words/get-unique-words.test.js]: I've updated this branch and amend tests to expect above error

**Pull request merged**